src/Voron/Data/BTrees/Tree.cs

```diff
@@ -1268,6 +1268,8 @@ private bool TryOverwriteOverflowPages(TreeNodeHeader* updatedNode, int len, out
                         _llt.FreePage(readOnlyOverflowPage.PageNumber + requestedOverflows + i);
```

Comment:
I'm pretty sure that we don't need `_llt.FreePage` now

Reply:
Yes, you do. Consider the case where you _shrink_ the value.
You wrote 64 KB, now you write 16 KB. We need a test for this, though

Reply:
I don't see a reason to call `FreePage` since we dropped `_dirtyOverflowPages` concept.

The initial 64KB write allocated page #10 (#23 in scratch). Now we have 16KB to write and we're in `TryOverwriteOverflowPages` optimization. We just need to discard #23, and allocate a new page in scratch for page #10 which will be smaller than initially.

Test:

```
[Theory]
        [InlineData(32 * 1024, 32 * 1024)]
        [InlineData(32 * 1024, 16 * 1024)]
        public void Can_update_big_value_in_the_same_transaction(int initialSize, int updateSize)
        {
            var r = new Random();

            using (var tx = Env.WriteTransaction())
            {
                Tree tree = tx.CreateTree("tree");

                var initial = new byte[initialSize];

                r.NextBytes(initial);
                tree.Add("key", initial);

                var updated = new byte[initialSize];

                r.NextBytes(updated);
                tree.Add("key", updated);

                tx.Commit();
            }
        }
```

Reply:
@ayende You added some tests to verify the behavior when updated value is shrink. Do you still think that `_llt.FreePage` is necessary here?

Reply:
Yes, inside `FreePage`, we are calling 

```
_freeSpaceHandling.FreePage(this, pageNumber);
_freedPages.Add(pageNumber);
```

Otherwise, we'll lose the space

Reply:
Right.